src/subcommand/init.rs

```diff
@@ -0,0 +1,104 @@
+//! Implementation for the `init` subcommand.
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+use crate::ext::{
+    error_stack::{ErrorHelper, IntoContext},
+    result::WrapOk,
+};
+use error_stack::Result;
+use indoc::formatdoc;
+
+/// Errors encountered during init.
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    /// A config file already exists
+    #[error("config file exists")]
+    ConfigFileExists,
+
+    /// Creating the data root directory
+    #[error("create data root")]
+    CreateDataRoot(PathBuf),
+
+    /// Writing the file did not work
+    #[error("write config file")]
+    WriteConfigFile(PathBuf),
+}
+
+/// generate the config and db files in the default location
+#[tracing::instrument(skip_all)]
+pub fn main(data_root: &PathBuf) -> Result<(), Error> {
```

Comment:
Why skip this arg?

Also: `Path` is to `PathBuf` what `str` is to `String`:
```suggestion
#[tracing::instrument(skip_all)]
pub fn main(data_root: &Path) -> Result<(), Error> {
```